src/librustc/mir/interpret/mod.rs

```diff
@@ -148,6 +148,8 @@ pub struct LitToConstInput<'tcx> {
 /// Error type for `tcx.lit_to_const`.
 #[derive(Copy, Clone, Debug, Eq, PartialEq, HashStable)]
 pub enum LitToConstError {
+    /// The literal's inferred type did not match the expected `ty` in the input.
+    TypeError,
```

Comment:
Maybe leave a comment that this only needs graceful handling in type-checking (which I didn't even realize was using this!), and everything else can ICE on it.